app/src/main/java/org/wikipedia/analytics/eventplatform/ImageRecommendationsEvent.kt

```diff
@@ -0,0 +1,41 @@
+package org.wikipedia.analytics.eventplatform
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import org.wikipedia.WikipediaApp
+
+@Serializable
+@SerialName("/analytics/mobile_apps/android_image_recommendation_event/1.0.0")
+class ImageRecommendationsEvent(
+   private val action: String,
+   private val active_interface: String,
+   private val action_data: String,
+   private val primary_language: String,
+   private val wiki_id: String
+) : MobileAppsEvent(STREAM_NAME) {
+
+    companion object {
+        private const val ACTION_IMPRESSION = "impression"
+        private const val STREAM_NAME = "eventlogging_EditAttemptStep"
+
+        fun logImpression(activeInterface: String) {
+            submitImageRecommendationEvent(ACTION_IMPRESSION, activeInterface, "", "")
+        }
+
+        fun logAction(action: String, activeInterface: String, actionData: String, wikiId: String) {
+            submitImageRecommendationEvent(action, activeInterface, actionData, wikiId)
+        }
+
+        fun getActionDataString(filename: String = "", recommendationSource: String = "", recommendationSourceProject: String = "",
+                                rejectionReasons: String = "", acceptanceState: String = "", seriesNumber: String = "",
+                                totalSuggestions: String = "", revisionId: String = "", captionAdd: String = "", altTextAdd: String = ""): String {
+            return "filename:$filename, recommendation_source:$recommendationSource,recommendation_source_project:$recommendationSourceProject, " +
+                    "rejection_reasons:$rejectionReasons, acceptance_state:$acceptanceState, series_number: $seriesNumber," +
+                    "total_suggestions: $totalSuggestions, revision_id:$revisionId, caption_add: $captionAdd, alt_text_add: $altTextAdd"
```

Comment:
It looks like `revision_id`, `caption_add`, and `alt_text_add` are not documented in the [Readme](https://gerrit.wikimedia.org/r/plugins/gitiles/schemas/event/secondary/+/refs/heads/master/jsonschema/analytics/mobile_apps/android_image_recommendation_event/README.md) of the schema. On the other hand, the Readme seems to document a `time_spent` parameter, which I'm not seeing here.

Reply:
[Done](https://gerrit.wikimedia.org/r/c/schemas/event/secondary/+/951181)